tools/config-generator/periodic_config.go

```diff
@@ -321,16 +328,44 @@ func generateGoCoveragePeriodic(title string, repoName string, _ yaml.MapSlice)
 			fmt.Sprintf("--cov-threshold-percentage=%d", data.Base.GoCoverageThreshold)}
 		data.Base.ServiceAccount = ""
 		data.Base.ExtraRefs = append(data.Base.ExtraRefs, "  base_ref: "+data.Base.RepoBranch)
-		if repositories[i].DotDev {
+		if repo.DotDev {
 			data.Base.ExtraRefs = append(data.Base.ExtraRefs, "  path_alias: knative.dev/"+path.Base(repoName))
 		}
-		if repositories[i].Go114 {
+		if repo.Go114 {
 			data.Base.SetGoVersion(GoVersion{1, 14})
 		}
 		addExtraEnvVarsToJob(extraEnvVars, &data.Base)
 		addMonitoringPubsubLabelsToJob(&data.Base, data.PeriodicJobName)
 		configureServiceAccountForJob(&data.Base)
 		executeJobTemplate("periodic go coverage", readTemplate(periodicCustomJob), title, repoName, data.PeriodicJobName, false, data)
-		return
+
+		betaData := data.Clone()
```

Comment:
A periodic job doesn't exercise the core logic of coverage job, the coverage diffing part, can we make this job somehow go through that code path to make sure?

Reply:
Not in any way we can display in TestGrid that I can think of

Reply:
The :coverage-dev job presubmit still exists

Reply:
Maybe we don't need it? this much custom logic in config generator makes me worry about maintenance

Reply:
The time for that ended a long time ago.

I want to have some sign coverage works with `prow-tests:beta`. This verifies at least half its functionality.

Reply:
Also, it's two function calls and a nice, clear, documented struct.

Reply:
I see line 342 to 369, like 28 lines of code rather than two function calls though

Reply:
Sure, but it's mostly naming things and static assignment of values. The only functions of consequence are a call to `executeJobTemplate` and `metaData.AddNonAlignedTest`.

We need some testing before releasing. This is decent and encompasses a lot of the functionality of the program, even if it isn't all of it.

Reply:
Well, I hate to block you because of this. Let's clean this mess up later